e2etests/tests/evpn_l2.go

```diff
@@ -90,108 +91,145 @@ var _ = Describe("Routes between bgp and the fabric", Ordered, func() {
+		nad, err = k8s.CreateMacvlanNad("110", testNamespace, "br-hs-110", tc.l2GatewayIPs)
 		Expect(err).NotTo(HaveOccurred())
-	})
 
-	AfterEach(func() {
-		removeLeafPrefixes(infra.LeafAConfig)
-		removeLeafPrefixes(infra.LeafBConfig)
-		dumpIfFails(cs)
-		err := Updater.CleanButUnderlay()
-		Expect(err).NotTo(HaveOccurred())
-		err = k8s.DeleteNamespace(cs, testNamespace)
-		Expect(err).NotTo(HaveOccurred())
-	})
+		DeferCleanup(func() {
+			removeLeafPrefixes(infra.LeafAConfig)
+			removeLeafPrefixes(infra.LeafBConfig)
+			dumpIfFails(cs)
+			err := Updater.CleanButUnderlay()
+			Expect(err).NotTo(HaveOccurred())
+			err = k8s.DeleteNamespace(cs, testNamespace)
+			Expect(err).NotTo(HaveOccurred())
+		})
 
-	It("should create two pods connected to the l2 overlay", func() {
-		var err error
-		const (
-			firstPodIP  = "192.171.24.2"
-			secondPodIP = "192.171.24.3"
-		)
 		nodes, err := k8s.GetNodes(cs)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(len(nodes)).To(BeNumerically(">=", 2), "Expected at least 2 nodes, but got fewer")
 
 		By("creating the pods")
-		firstPod, err = k8s.CreateAgnhostPod(cs, "pod1", testNamespace, k8s.WithNad(nad.Name, testNamespace, firstPodIP+"/24"), k8s.OnNode(nodes[0].Name))
+		firstPod, err = k8s.CreateAgnhostPod(cs, "pod1", testNamespace, k8s.WithNad(nad.Name, testNamespace, tc.firstPodIPs), k8s.OnNode(nodes[0].Name))
 		Expect(err).NotTo(HaveOccurred())
-		secondPod, err = k8s.CreateAgnhostPod(cs, "pod2", testNamespace, k8s.WithNad(nad.Name, testNamespace, secondPodIP+"/24"), k8s.OnNode(nodes[1].Name))
+		secondPod, err = k8s.CreateAgnhostPod(cs, "pod2", testNamespace, k8s.WithNad(nad.Name, testNamespace, tc.secondPodIPs), k8s.OnNode(nodes[1].Name))
 		Expect(err).NotTo(HaveOccurred())
 
 		By("removing the default gateway via the primary interface")
-		removeGatewayFromPod(firstPod)
-		removeGatewayFromPod(secondPod)
+		Expect(removeGatewayFromPod(firstPod)).To(Succeed())
+		Expect(removeGatewayFromPod(secondPod)).To(Succeed())
 
 		checkPodIsReacheable := func(exec executor.Executor, from, to string) {
-			Eventually(func() error {
+			GinkgoHelper()
+			const port = "8090"
+			from = strings.Split(from, "/")[0]
+			to = strings.Split(to, "/")[0]
+			hostPort := net.JoinHostPort(to, port)
+			urlStr := url.Format("http://%s/clientip", hostPort)
+			Eventually(func(g gomega.Gomega) string {
 				By(fmt.Sprintf("trying to hit %s from %s", to, from))
-				urlStr := url.Format("http://%s:8090/clientip", to)
 				res, err := exec.Exec("curl", "-sS", urlStr)
-				if err != nil {
-					return fmt.Errorf("curl %s:8090 failed: %s", to, res)
-				}
-				clientIP := strings.Split(res, ":")[0]
-				if clientIP != from {
-					return fmt.Errorf("curl %s:8090 returned %s, expected %s", to, clientIP, from)
-				}
-
-				return nil
-			}, time.Minute, 5*time.Second).ShouldNot(HaveOccurred())
+				g.Expect(err).ToNot(HaveOccurred(), "curl %s failed: %s", hostPort, res)
```

Comment:
this won't take into account the time for the network to be setup. The first time the curl will fail the test will fail too

Reply:
ahhh sorry I missed that, thanks!